src/haggadah/views.cljs

```diff
@@ -122,40 +122,101 @@
          [:g {:transform "translate(-4.000000, 76.000000)", :fill "#FFFFFF", :fill-rule "nonzero"}
           [:path {:d "M0.457,34.035 C57.086,53.198 98.208,65.809 123.822,71.865 C181.454,85.495 234.295,90.29 272.033,93.459 C311.355,96.759 396.635,95.801 461.025,91.663 C486.76,90.01 518.727,86.372 556.926,80.752 C595.747,74.596 622.372,70.008 636.799,66.991 C663.913,61.324 712.501,49.503 727.605,46.128 C780.47,34.317 818.839,22.532 856.324,15.904 C922.689,4.169 955.676,2.522 1011.185,0.432 C1060.705,1.477 1097.39,3.129 1121.236,5.387 C1161.703,9.219 1208.621,17.821 1235.4,22.304 C1285.855,30.748 1354.351,47.432 1440.886,72.354 L1441.191,104.352 L1.121,104.031 L0.457,34.035 Z"}]]]]])
 
+(defn icon
+  "Pre: takes two Haggadah ids
+  Post: returns a filled circle icon if the ids match. Returns a unfilled circle icon otherwise"
+  [id-1 id-2]
+  (if (= id-1 id-2)
+    [:i.fa-circle.fas]
+    [:i.fa-circle.far]))
+
+(defn all-haggadot
+  "Pre: takes a collection of Haggadot and a selected Haggadah
+  Post: returns a hiccup representation of a collection of Haggadot"
+  [haggadot]
+   (let [haggadah-picked @(re-frame/subscribe [::subs/selected-haggadah])]
+     [:div.dropdown-content
+      (for [{:keys [title id]} haggadot :when id]
+        ^{:key id}
+        [:a.dropdown-item.seder-option
+         {:on-click #(re-frame/dispatch [::events/selected-haggadah id])
```

Comment:
Perhaps having a utility function (that returns a function) here would make this look a bit better like

```clojure
[a.dropdown ... {:on-click (dispatch ::events/selected-haggadah id) ... }
```

Reply:
Or perhaps also a function that returns a `Map`.

```clj
[:a.dropdown... (on-click-dispatch ::events/selected-haggadah id)]
```

Reply:
The pattern of dispatching an event vector is repeatedly used. I'll make a utility function which can return the dispatching function.